src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -496,7 +500,7 @@ public static bool IsCollectionAddMethod(this IMethodSymbol method, ImmutableHas
         /// <param name="taskType">Task type.</param>
         public static bool IsTaskFromResultMethod(this IMethodSymbol method, [NotNullWhen(returnValue: true)] INamedTypeSymbol? taskType)
             => method.Name.Equals("FromResult", StringComparison.Ordinal) &&
-               method.ContainingType.Equals(taskType);
+               SymbolEqualityComparer.Default.Equals(method.ContainingType, taskType);
```

Comment:
Shall we check the `OriginalDefinition` here?

Reply:
> Shall we check the `OriginalDefinition` here?

This is the non-generic task type. So I think the way it's currently written is okay.